src/app/event/[id]/EventDetail.tsx

```diff
@@ -24,13 +43,23 @@ export const EventDetail: React.FC<EventDetailProps> = ({
       )}
     >
       <article className="flex flex-col rounded-md bg-white">
-        <HeartButton className="ml-auto" />
+        {userId === -1 ? (
```

Comment:
We should return null if the id is not found. If it would create a lot of problems, create a constant with -1 in it